source/draw/RasterMonochrome.ooc

```diff
@@ -43,25 +43,56 @@ RasterMonochrome: class extends RasterPacked {
 		if (monochrome != image)
 			monochrome referenceCount decrease()
 	}
-	// Precondition: Do not draw out of bound
-	_blitWhiteRaw: static func (target, source: Byte*, startSrcX, startSrcY, startDstX, startDstY, width, height, targetStride, sourceStride: Int) {
-		for (y in 0 .. height) {
-			dstY := startDstY + y
-			srcY := startSrcY + y
-			writer := target + startDstX + (dstY * targetStride)
-			reader := source + startSrcX + (srcY * sourceStride)
-			for (x in 0 .. width) {
-				sourceColor: Float = reader@
-				targetColor: Float = writer@
-				blendColor: Int = (targetColor * (1.0f - (sourceColor / 255.0f))) + sourceColor // Source intensity as alpha
-				if (blendColor > 255) blendColor = 255 // Limit intensity
-				writer@ = blendColor // Write to target
-				reader += 1
-				writer += 1
+	// Preconditions:
+	//   Do not draw out of bound.
+	//   Do not let the target region overlap the source region.
+	_blitRaw: static func (target, source: Byte*, startSrcX, startSrcY, startDstX, startDstY, width, height, targetStride, sourceStride: Int, blendMode: BlendMode) {
+		if (width > 0) {
+			upperLeftTarget := target + startDstX + (startDstY * targetStride)
+			upperLeftSource := source + startSrcX + (startSrcY * sourceStride)
+			writerLineStart := upperLeftTarget
+			readerLineStart := upperLeftSource
+			match (blendMode) {
+				case BlendMode White =>
+					for (y in 0 .. height) {
+						writer := writerLineStart
```

Comment:
I would have to read the assembly output and check if the C compiler inlines them since C compilers are very careful about inlining.